component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/token/binder/DPoPBasedTokenBinder.java

```diff
@@ -76,7 +78,7 @@ public String getBindingType() {
     @Override
     public List<String> getSupportedGrantTypes() {
 
-        return Collections.unmodifiableList(supportedGrantTypes);
+        return new ArrayList<>(Arrays.asList(getAllGrantTypes()));
```

Comment:
How can we set all the grant types as supported? We are supporting different grant types OOTB. Did we test DPoP with all the grant types

Reply:
I have tested DPoP with the below grant types.

1. Authorization Code Grant.
2. Password Grant.
3. Client Credentials Grant.
4. Refresh Token Grant.
5. Custom grant type[1].

[1] - https://is.docs.wso2.com/en/latest/learn/writing-a-custom-oauth-2.0-grant-type/#customizing-an-existing-grant-type